### PR TITLE
Data mate functions

### PR DESCRIPTION
**Pull request by jsnoble**
- [x] Add Cast function config
- [x] Fix SetTimezone to take in a timezone as well as an offset
- [ ] Add tests for timezoneToOffset

### CODE REVIEW

File: packages/data-mate/test/function-configs/function-configs-spec.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ import { functionTestHarness } from './functionTestHarness';
 
 describe('function configs', () => {
     Object.entries(functionConfigRepository).forEach(([key, fnDef]) => {
-        functionTestHarness(fnDef as FunctionDefinitionConfig<any>, key);
+        if (key === 'setTimezone') functionTestHarness(fnDef as FunctionDefinitionConfig<any>, key);
```
Comment (peterdemartini) [inline packages/data-mate/test/function-configs/function-configs-spec.ts:13]:
```suggestion
        functionTestHarness(fnDef as FunctionDefinitionConfig<any>, key);
```

Comment (peterdemartini) [reply packages/data-mate/test/function-configs/function-configs-spec.ts:13]: I have seen you leave this in a few times, I strongly encourage you to use the `-t` flag when running the test so, `yarn test:debug test/function-configs/function-configs-spec.ts -t 'setTimezone'` or `yarn test:watch test/function-configs/function-configs-spec.ts -t 'setTimezone'`. Running tests via jest watch is very powerful and it is very quick to add additional filters by test name, run only the failed tests, or switch to a different file, just type `w` when all the tests have complete in watch mode for help options

**Comment by jsnoble** [comment]
I am having a hard time coming up with a way to tests this function as it innately relies on side effects that change over the span of a year, and changes based off of where the tests are being run.

This will become a little more reliable when the servers are run in Arizona, but the CI and others could run this test in any place as this is open source. I am not sure if there is a proper way to tests this function. 

Also, our interface to it is very simplistic, so it would mainly be testing the function from another library itself, which I am not sure is needed

**Review by peterdemartini — APPROVED** [review]
LGTM

**Pull request merged**